### PR TITLE
Avoid using dype char codes

### PR DESCRIPTION
**Pull request by niboshi**
NumPy dtype char codes (such as `'f'`, `'i'`) are expressly discouraged in NumPy documentation:
https://docs.scipy.org/doc/numpy/reference/arrays.scalars.html#built-in-scalar-types

They represent corresponding C types (such as `float` and `int`), whose actual bit widths may be dependent on environment.

### CODE REVIEW

**Comment by toslunar** [comment]
See also #4350, which changed char codes in the documents and the examples.

**Comment by toslunar** [comment]
Jenkins, test this please.

File: chainer/links/connection/convolution_nd.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ class ConvolutionND(link.Link):
 
         Let an input vector ``x`` be:
 
-        >>> x = np.arange(2 * 5 * 5 * 5, dtype='f').reshape(1, 2, 5, 5, 5)
+        >>> x = np.arange(2 * 5 * 5 * 5, dtype=numpy.float32).reshape(1, 2, 5, 5, 5)
```
Comment (kmaehashi) [inline chainer/links/connection/convolution_nd.py]: s/numpy/np/

File: chainer/links/connection/deconvolution_nd.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ class DeconvolutionND(link.Link):
 
         Let an input vector ``x`` be:
 
-        >>> x = np.arange(2 * 5 * 5 * 5, dtype='f').reshape(1, 2, 5, 5, 5)
+        >>> x = np.arange(2 * 5 * 5 * 5, dtype=numpy.float32).reshape(1, 2, 5, 5, 5)
```
Comment (kmaehashi) [inline chainer/links/connection/deconvolution_nd.py]: ditto

File: chainer/functions/activation/rrelu.py [hunk 3]
```diff
@@ -126,7 +126,7 @@ def rrelu(x, l=1. / 8, u=1. / 3, **kwargs):
 
     .. admonition:: Example
 
-        >>> x = np.array([[-1, 0], [2, -3], [-2, 1]], 'f')
+        >>> x = np.array([[-1, 0], [2, -3], [-2, 1]], numpy.float32)
```
Comment (kmaehashi) [inline chainer/functions/activation/rrelu.py]: s/numpy/np/

**Comment by niboshi** [comment]
Thank you for pointing out.
Fixed as suggested.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**